File: integration_tests/java/user/PreferencesDoNotTrackV1DoNotTrack.java
package com.jsontypedef.jtdcodegendemo;

/**
 * A multi-level do-not-track setting
 */
public enum PreferencesDoNotTrackV1DoNotTrack {

  /**
   * All forms of tracking permitted.
   */
  @JsonProperty("ALL")
  ALL,

  /**
   * Only essentialy forms of tracking permitted.
   */
  @JsonProperty("ESSENTIAL_ONLY")
  ESSENTIAL_ONLY,

  /**
   * No forms forms of tracking permitted.
   */
  @JsonProperty("NONE")
  NONE,

}

File: integration_tests/csharp/user/PreferencesDoNotTrackV1DoNotTrack.cs
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Jtd.JtdCodegenDemo
{
    /// <summary>
    /// A multi-level do-not-track setting
    /// </summary>
    [JsonConverter(typeof(StringEnumConverter))]
    public enum PreferencesDoNotTrackV1DoNotTrack
    {

        /// <value>
        /// All forms of tracking permitted.
        /// </value>
        [EnumMember(Value = "ALL")]
        All,

        /// <value>
        /// Only essentialy forms of tracking permitted.
        /// </value>
        [EnumMember(Value = "ESSENTIAL_ONLY")]
        EssentialOnly,

        /// <value>
        /// No forms forms of tracking permitted.
        /// </value>
        [EnumMember(Value = "NONE")]
        None,

    }
}
